Project.toml

```diff
@@ -20,6 +20,7 @@ LogDensityProblems = "6fdf6af0-433a-55f7-b3ed-c6c6e0b8df7c"
 MCMCChains = "c7f686f2-ff18-58e9-bc7b-31028e88f75d"
 MacroTools = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
 Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
+Memoization = "6fafb56a-5788-4b4e-91ca-c0cea6611c73"
```

Comment:
I just noticed that it's a bit strange that ReverseDiff is an optional dependency but Memoization which is only used for ReverseDiff is not. Maybe make it an optional dependency as well, nested inside of the `@requires` block for ReverseDiff?

Reply:
I don't think Requires supports this though afaiu. Does it?

Reply:
Yes, it does: https://github.com/JuliaPackaging/Requires.jl#multiple-requirements

Reply:
But technically Memoization is not independently optional. So the user would have to load it too.

Reply:
`using ReverseDiff, Turing` would not be enough.

Reply:
Sure, but that makes sense, doesn't it? As long as the user does not load Memoization, such as in
```
using Turing
using ReverseDiff
```
the tape is not cached. But an additional `using Memoization` would enable support for caching the tape. Since ReverseDiff is optional at the moment and caching has to be enabled anyway, that's quite reasonable, I guess?

Reply:
Sure, sounds good, as long as all is well documented.